calculator.rb

```diff
@@ -0,0 +1,131 @@
+# Ada C12 - Tiffany Chio
+# This is the long way of making Ruby perform as a calculator.
+# The program prompts for an operator and two operands.
+# Then prints the answer.
+
+# Validation function for the operator
+# The user input must match one of pre-defined keywords/symbols.
+# Will print error message and prompt for a new input until a valid one is obtained.
+def val_operation(operation)
+  valid = false
+  while valid == false
+    if ["add", "+", "subtract", "-", "multiply", "*", "divide", "/", "exponent", "^", "modulo", "%"].include? operation
+      valid = true
+    else
+      puts ""
+      puts "That is not a valid operator. Please try again."
+      print "Please choose one operator(name or symbol): "
+      operation = gets.chomp.downcase
+    end
+  end
+  return operation
+end
+
+# Operand validation function.
+# Checking that the input is a valid number.
+# Will print error message and prompt for a new input until a valid one is obtained.
+# Will also return either int or float as appropriate.
+def val_number(number, prompt)
+  valid = false
+
+  # Function determines if the user entered an int or float by checking
+  # for the prescence of a decimal point.
+  # Converts string input into appropriate class and returns that value.
+  # If...the number is a float
+  if number.match?(/\.+/)
```

Comment:
Often it can be useful to have a boolean such as valid = true. However, in this case it seems like you could simply have a while loop without the tests to simplify your code.

```
while number.to_f.to_s == number
....
```